tools/windows/DatadogAgentInstaller/CustomActions/UserCustomActions.cs

```diff
@@ -156,6 +166,122 @@ public static ActionResult ReadRegistryProperties(Session session)
             return ReadRegistryProperties(new SessionWrapper(session));
         }
 
+        /// <summary>
+        /// Returns true if we will treat <c>name</c> as an alias for the local machine name.
+        /// </summary>
+        /// <remarks>
+        /// Comparisons are case-insensitive.
+        /// </remarks>
+        private static bool NameIsLocalMachine(string name)
+        {
+            name = name.ToLower();
+            if (name == Environment.MachineName.ToLower())
+            {
+                return true;
+            }
+            // Windows runas does not support the following names, but we can try to.
+            if (name == ".")
+            {
+                return true;
```

Comment:
TBH I was a little torn about including `"."` here at all. Because on the DC we substitute `"."` for the domain name, not the machine name. So in that context it's not accurate to say `"."` is an alias for the local machine. I made `NameUsesDefaultPart` to keep them separate in that way.